posthog/api/test/test_feature_flag.py

```diff
@@ -812,6 +812,43 @@ def test_updating_a_feature_flag_with_same_team_and_key_of_a_deleted_one(self):
         instance.refresh_from_db()
         self.assertEqual(instance.key, "alpha-feature")
 
+    def test_my_flags_is_not_nplus1(self):
+
+        query_counts: List[int] = []
+        queries: List[List[Dict[str, str]]] = []
+
+        count, qs = self._get_flags_counting_queries()
+        query_counts.append(count)
+        queries.append(qs)
+
+        # add new flags
+        for i in range(5):
+            self.client.post(
+                f"/api/projects/{self.team.id}/feature_flags/",
+                data={"name": f"flag-{i}", "key": f"flag-{i}", "filters": {"groups": [{"rollout_percentage": i,}]},},
+                format="json",
+            ).json()
+
+            count, qs = self._get_flags_counting_queries()
+            query_counts.append(count)
+            queries.append(qs)
+
+        # query counts don't climb as flags are added
+        self.assertTrue(
+            all(j - i < 2 for i, j in zip(query_counts[0:], query_counts[1:])),
+            f"received: {query_counts} query counts when loading my_flags",
+        )
+
+    def _get_flags_counting_queries(self) -> Tuple[int, List[Dict[str, str]]]:
+        db_connection = connections[DEFAULT_DB_ALIAS]
+
+        with CaptureQueriesContext(db_connection) as capture_query_context:
```

Comment:
See my suggestion above, but I think we want to avoid being clever here and use `with self.assertNumQueries(expected_query_count):` instead.

Reply:
🤔 we're using this in a few places now... my next step was to lift it out to reduce the duplication. 

I'll follow-up and we can either standardise this or replace it but in all three or so places

Reply:
This is only used once in this PR though?

Reply:
The "drawback" with `assertNumQueries` is that the query count could change for reasons outside of this code. We don't care about the number of queries necessarily but the difference between repeated executions

Reply:
> The "drawback" with assertNumQueries is that the query count could change for reasons outside of this code. 

I would say this is a positive - if we add extra queries to this timing-sensitive endpoint, let it fail and it will be explicit in the PR that adds the issue.

As I mention above though this test is too clever right now and hard to debug/maintain for that reason. Keeping to a standard SETUP - RUN - ASSERT structure keeps this more consistent with the rest of the codebase.

Reply:
I don't agree that a loop in a test is hard to maintain... But I don't feel particularly strongly about _this_ test.

Pushed a change... is that what you meant would be clearer?